### PR TITLE
Switch RMSE to MSE (true L2 loss)

### PR DESCRIPTION
**Pull request by Laurae2**
To keep consistency with the name "Mean Squared Error" as it is not RMSE.

We may add RMSE back under a better name in another PR, because L2 = MSE being RMSE is causing confusion for users.

### CODE REVIEW

**Comment by guolinke** [comment]
@Laurae2 can you keep both RMSE and MSE, not just replace it ?

**Comment by Laurae2** [comment]
@guolinke I can try keeping both at the same time. It works it seems!

image

```r
library(lightgbm)
data(agaricus.train, package = "lightgbm")
train <- agaricus.train
dtrain <- lgb.Dataset(train$data, label = train$label)
data(agaricus.test, package = "lightgbm")
test <- agaricus.test
dtest <- lgb.Dataset.create.valid(dtrain, test$data, label = test$label)
valids <- list(test = dtest)

params <- list(objective = "regression", metric = "rmse", device = "cpu", nthread = 1)
model <- lgb.train(params,
                   dtrain,
                   10,
                   valids,
                   min_data = 1,
                   learning_rate = 1,
                   early_stopping_rounds = 10)
rmse <- as.numeric(model$record_evals$test$rmse$eval)

params <- list(objective = "regression", metric = "mse", device = "cpu", nthread = 1)
model <- lgb.train(params,
                   dtrain,
                   10,
                   valids,
                   min_data = 1,
                   learning_rate = 1,
                   early_stopping_rounds = 10)
mse <- as.numeric(model$record_evals$test$l2$eval)

plot(rmse, mse)
curve(x^2, from = 0, to = 0.03, add = TRUE)
```
